src/Link.js

```diff
@@ -39,7 +40,13 @@ Link.defaultProps = {
 }
 
 Link.propTypes = {
-  as: PropTypes.oneOf(['a', 'button', 'input', 'summary']),
+  as: (props, propName, componentName) => {
+    if (props[propName] && !isValidElementType(props[propName])) {
+      return new Error(
+        `Invalid prop '${propName}' supplied to '${componentName}': the prop is not a valid React component`
+      )
+    }
+  },
```

Comment:
This function should probably be pulled out into the `utils` folder so it can be used to define propTypes for other components. I'm not sure what to call the function though. Any ideas @emplums @shawnbot?

Reply:
Agree, let's move this into Utils! Maybe call it `isValidComponent`?